examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -59,6 +59,18 @@ CHIP_ERROR PairingCommand::RunCommand()
     return RunInternal(mNodeId);
 }
 
+void PairingCommand::CancelCommand(CHIP_ERROR status)
+{
+    ChipLogProgress(chipTool, "Cancel pairing: %s", chip::ErrorStr(status));
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+    if ((mPairingMode == PairingMode::WiFiPAF) || (mPairingMode == PairingMode::Code))
+    {
+        ChipLogProgress(chipTool, "Cancel pairing, node: %lu", mNodeId);
+        chip::WiFiPAF::WiFiPAFLayer::GetWiFiPAFLayer().CancelSession(mNodeId, chip::WiFiPAF::kWiFiPafRole_Subscriber);
```

Comment:
Why are we touching PAF stuff manually here?  Shouldn't this just happen during normal stack shutdown if we just go ahead and shut down the stack cleanly?

Reply:
If we did not shutdown stack, we need to cancel the subscription we did in the previous pairing command.

Reply:
How does this work for "normal" pairing when pairing is stopped without shutting down the stack? 

It feels like we are adding a chip-tool specific fix for what is a core problem here...

Reply:
In the normal condition, commissionee will shutdown the PAF session while the commissioning is completed. Then if the session has started and no traffic for a long time, the commissioner will also shutdown the PAF session.
I will add to cancel the command after it's done to be more explicitly.

Reply:
chip-tool is the commissioner, not the commissionee.  So in the "normal condition", what cancels the PAF stuff on the commissioner side if commissioning is interrupted?

Reply:
There is a time-out if the traffic stops too long that it will cancel the PAF. I added to cancel PAF once it's completed yesterday in the PR.

Reply:
I guess I am basically being confused about why the commissioner in chip-tool needs special logic here that is different from "normal" commissioners.  It feels like we are trying to special-case chip-tool somehow when the right fix should be somewhere deeper....

Reply:
I will add an API, similar with CloseBleConnection() in DeviceCommissioner and move the code there.